libvisual/libvisual/lv_video.cpp

```diff
@@ -259,6 +263,21 @@ namespace LV {
       return m_impl->buffer->is_allocated ();
   }
 
+  bool Video::save_to_file (std::string const& path) const
+  {
+      auto extension = fs::path {path}.extension ();
+
+      std::ofstream file {path};
```

Comment:
I believe this^^ line needs to be moved into the `if (extension == ".png")` block 0r else an empty file will have been created already.

Reply:
@hartwork, nice catch. I was already tackling this albeit indirectly, thinking of the scenario where the file creation failed instead.

Reply:
@hartwork This code was only just ported from a graphics engine I'm writing on the side. The port is still untested.

At the moment, I'm trying to build up the testing support a little so I can write unit tests for this and other `LV::Video` functions. In the works are two methods for `LV::Video`, `validate()` and `has_same_content()` for validating the internal consistency of an instance, and for comparing two instances respectively.

Reply:
> @hartwork This code was only just ported from a graphics engine I'm writing on the side. The port is still untested.

@kaixiong how do you feel about using either [draft status](https://github.blog/2019-02-14-introducing-draft-pull-requests/) or a subject prefix like "[W.I.P.] " for cases like this? I think that that would make it quicker to see (without need to remember) for cases like this.

Reply:
@hartwork Done. This was the intention but I forgot to set the draft status it seems.